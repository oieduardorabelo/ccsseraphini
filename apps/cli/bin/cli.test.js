let path = require('path');
let cp = require('child_process');

let cli = path.join(__dirname, './cli.js');

describe('cc @sseraphini / cli', () => {
  it('show help', () => {
    let output1 = setup();
    let output2 = setup(['-h']);
    let output3 = setup(['--help']);
    expect(output1).toMatchSnapshot();
    expect(output2).toMatchSnapshot();
    expect(output3).toMatchSnapshot();
  });

  it('show version', () => {
    let output1 = setup(['-v']);
    let output2 = setup(['--version']);
    expect(output1).toMatchSnapshot();
    expect(output2).toMatchSnapshot();
  });

  it('show error', () => {
    let output1 = setup(['-bla']);
    expect(output1).toMatchSnapshot();
  });

  it('opens url with text', async () => {
    let output1 = cp.spawn(cli, ['how can i tweet']);
    await new Promise((res) => {
      output1.on('close', (code) => {
        expect(code).toBe(0); // we assume code exit 0 is a success
        res();
      });
    });
  });
});

function setup(args = [], options = {}) {
  return cp.spawnSync('node', [cli, ...args], { encoding: 'utf-8', ...options })
    .stdout;
}
