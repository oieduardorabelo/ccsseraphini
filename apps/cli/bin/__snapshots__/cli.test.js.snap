// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cc @sseraphini / cli show error 1`] = `
"[31mError[0m: Found argument '-bla' which wasn't expected, or isn't valid in this context!

[33mUSAGE[0m:
  npx ccsseraphini [QUESTION]

For more information try --help or -h
"
`;

exports[`cc @sseraphini / cli show help 1`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE[0m:
  npx ccsseraphini [QUESTION]

[33mEXAMPLE[0m:
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS[0m:
  -h, --help                           Print this help message.
  -V, --version                        Show version information.
"
`;

exports[`cc @sseraphini / cli show help 2`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE[0m:
  npx ccsseraphini [QUESTION]

[33mEXAMPLE[0m:
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS[0m:
  -h, --help                           Print this help message.
  -V, --version                        Show version information.
"
`;

exports[`cc @sseraphini / cli show help 3`] = `
"[32mcli[0m [32m0.0.1[0m
ccsseraphini is a CLI to solve all your doubts about everything.

[33mUSAGE[0m:
  npx ccsseraphini [QUESTION]

[33mEXAMPLE[0m:
  npx ccsseraphini \\"How can I learn Relay?\\"

[33mOPTIONS[0m:
  -h, --help                           Print this help message.
  -V, --version                        Show version information.
"
`;

exports[`cc @sseraphini / cli show version 1`] = `
"cli 0.0.1
"
`;

exports[`cc @sseraphini / cli show version 2`] = `
"cli 0.0.1
"
`;
